ilc/server/tailor/configs-injector.spec.js

```diff
@@ -0,0 +1,429 @@
+            const browserTimingHeader = `<script defer type="text/javascript">window.browserTimingHeader = 'Hi there! I should add a timing header.';</script>`;
+            newrelic.getBrowserTimingHeader.withArgs().returns(browserTimingHeader);
+
+            const configsInjector = new ConfigsInjector(newrelic);
+
+            const request = {};
+
+            const template = {
+                styleRefs: [],
+                content:
+                    '<html>' +
+                        '<head>' +
+                            '<!-- ILC_JS -->' +
+                            '<title>Configs Injector`s test</title>' +
+                            '<!-- ILC_CSS -->' +
+                            '<link rel="stylesheet" href="https://somewhere.com/style.css">' +
+                            '<!-- ILC_JS -->' +
+                            '<!-- ILC_CSS -->' +
+                        '</head>' +
+                        '<body>' +
+                            '<!-- ILC_JS -->' +
+                            '<div>Hi there! I am content.</div>' +
+                            '<!-- ILC_CSS -->' +
+                        '</body>' +
+                    '</html>',
+            };
+
+            chai.expect(configsInjector.inject(request, registryConfig, template, slots)).to.be.eql(
+                '<html>' +
+                    '<head>' +
+                        '<!-- TailorX: Ignore during parsing START -->' +
+                        `<script type="spa-config">${JSON.stringify({
+                            apps: {
+                                firstApp: {
+                                    spaBundle: registryConfig.apps.firstApp.spaBundle,
+                                    cssBundle: registryConfig.apps.firstApp.cssBundle,
+                                    dependencies: {
+                                        ...registryConfig.apps.firstApp.dependencies,
+                                    },
+                                    props: {
+                                        ...registryConfig.apps.firstApp.props,
+                                    },
+                                    kind: registryConfig.apps.firstApp.kind,
+                                },
+                                secondApp: {
+                                    spaBundle: registryConfig.apps.secondApp.spaBundle,
+                                    cssBundle: registryConfig.apps.secondApp.cssBundle,
+                                    dependencies: {
+                                        ...registryConfig.apps.secondApp.dependencies,
+                                    },
+                                    props: {
+                                        ...registryConfig.apps.secondApp.props,
+                                    },
+                                    kind: registryConfig.apps.secondApp.kind,
+                                },
+                                thirdApp: {
```

Comment:
Couldn't you just write `thirdApp: registryConfig.apps.thirdApp`?

Reply:
Maybe use _.pick() if you need only some props

Reply:
I changed it, so I created a function that should return SPA config. _.pick is used to get props of apps there.